### PR TITLE
Fix: dynamically adapt pose loss names to model type

### PR DESCRIPTION
**Pull request by lmycross**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Improves loss reporting for YOLO pose training by only logging `rle_loss` when it’s actually supported, plus clarifies loss component meanings 🧾✅

### 📊 Key Changes
- Conditionally appends `rle_loss` to pose trainer `loss_names` **only if** the model head includes a `flow_model` (i.e., when RLE is available) 🔁
- Updates comments in `ultralytics/utils/loss.py` to better reflect what returned loss tensors represent (including keypoint-related losses) 📝
- Clarifies the oriented bounding box (OBB) loss vector includes an **angle** component (`box, cls, dfl, angle`) 🧭

### 🎯 Purpose & Impact
- Prevents confusing logs/metrics by **not advertising `rle_loss` when it’s not used**, making training output more accurate and easier to interpret 📉➡️📈
- Helps users and developers understand what each loss term corresponds to, improving debugging and readability 🧠🔍
- Reduces risk of mismatched expectations in training dashboards, callbacks, or custom tooling that relies on `loss_names` 📊🛠️

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lmycross, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message, and an engineer will assist shortly. To ensure a seamless integration of your work, please review the following checklist:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/models/yolo/pose/train.py [hunk 1]
```diff
@@ -90,7 +90,12 @@ def set_model_attributes(self):
 
     def get_validator(self):
         """Return an instance of the PoseValidator class for validation."""
-        self.loss_names = "box_loss", "pose_loss", "kobj_loss", "cls_loss", "dfl_loss", "rle_loss"
+        head = self.model.model[-1]
```
Comment (UltralyticsAssistant) [inline ultralytics/models/yolo/pose/train.py]: ⚠️ **HIGH**: `self.loss_names` is now sometimes 5 items, but the pose loss vector in `ultralytics/utils/loss.py` still appears to be length 6 (it unconditionally indexes `loss[5]` and only gates scaling by `self.rle_loss is not None`). If downstream logging/validators zip names with a 6-element loss tensor, this change can reintroduce the very mismatch it’s trying to fix (5 names vs 6 values). Consider either always keeping `"rle_loss"` in `loss_names` (it will just stay 0 when RLE is absent), or making the loss tensor length truly dynamic when RLE is disabled.

File: ultralytics/utils/loss.py [hunk 2]
```diff
@@ -834,7 +834,7 @@ def loss(self, preds: dict[str, torch.Tensor], batch: dict[str, torch.Tensor]) -
         if self.rle_loss is not None:
             loss[5] *= self.hyp.rle  # rle gain
 
-        return loss * batch_size, loss.detach()  # loss(box, cls, dfl)
+        return loss * batch_size, loss.detach()  # loss(box, cls, dfl, kpt_location, kpt_visibility)
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/loss.py:837]: 💡 **MEDIUM**: The updated inline comment doesn’t match the actual loss vector indexing shown a few lines above (`loss[5]` is used for RLE). As written, `# loss(box, cls, dfl, kpt_location, kpt_visibility)` omits the optional `rle` component and still doesn’t reflect the full ordering used elsewhere (e.g., box/pose/kobj/cls/dfl/rle). This can mislead future edits/debugging.

**Suggested change:**
```suggestion
        return loss * batch_size, loss.detach()  # loss(box, pose, kobj, cls, dfl, rle)
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good intent and the head capability check is straightforward, but there’s a likely correctness issue: the pose loss tensor still looks 6-wide (RLE slot present even when disabled), while `loss_names` is sometimes reduced to 5. Also one inline loss comment in `loss.py` is now inconsistent with the actual loss indexing.

💬 Posted 2 inline comments

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged — huge thanks @lmycross (and @Laughing-q for the added contributions) for this thoughtful improvement. By conditionally logging `rle_loss` only when RLE is actually supported and sharpening the inline loss-component descriptions (including the OBB angle term), you’ve made YOLO pose training outputs more truthful, easier to debug, and safer for anyone building dashboards/callbacks around `loss_names`.

As Florence Nightingale put it, *“I attribute my success to this: I never gave or took any excuse.”* This PR embodies that mindset—no “phantom” metrics, just clear, accountable reporting that helps the whole community train with confidence.
